common/util/Assert.h

```diff
@@ -3,16 +3,23 @@
  * Custom ASSERT macro
  */
 
+#include "third-party/fmt/core.h"
```

Comment:
can we remove this? I'd rather not include `libfmt` in more places than we need because it's a huge header-only library that increases compile times. I think we can just put `#include <string>` instead.